tools/mason/MasonPublish.chpl

```diff
@@ -0,0 +1,248 @@
+      throw new owned MasonError('Package does not gave a git origin');
+    }
+  }
+}
+
+
+proc moveToMasonHome() {
+  const masonHome = getEnv("MASON_HOME");
+  here.chdir(masonHome);
+}
+
+/* Clones the mason registery fork from the users repo. Takes username as input. */
+proc cloneMasonReg(username: string) throws {
+  try! {
+    var ret = runCommand("git clone git@github.com:" + username + "/mason-registry mason-registry", true);
+    return ret;
+    }
+  catch {
+    writeln('Error cloning the fork of mason-registry');
+    writeln('Make sure you have forked the mason-registry on GitHub');
+    exit(1);
+    }
+}
+
+/* Checks to see if 'git config --get remote.origin.url' exists */
+proc doesGitOriginExist() {
+  var urlExists = runCommand("git config --get remote.origin.url", true);
+  if urlExists != '' {
+    return true;
+  }
+  else {
+    return false;
+  }
+}
+
+/*Procedure that reutns the url of the git remote origin */
+proc gitUrl() {
+  var url = runCommand("git config --get remote.origin.url", true);
+  return url;
+}
+
+/* Takes the git username and creates a new branch of the mason registry users fork,
+ name or branch is taken from the Mason.toml of the mason package.*/
+proc branchMasonReg(username: string, name: string) throws {
+  try! {
+    const masonReg = "/mason-registry/";
+    const branchCommand = "git checkout -b "+ name: string;
+    var ret = gitC(MASON_HOME + '/tmp/mason-registry', branchCommand, true);
+    return ret;
+    }
+  catch {
+    here.chdir(MASON_HOME);
+    rmTree('/tmp/');
+    writeln('Error branching mason-registry');
+    exit(1);
+    }
+}
+
+/* Gets name from the Mason.toml */
+proc getName() throws {
+  try! {
+    const cwd = getEnv("PWD");
+    const projectHome = getProjectHome(cwd);
+    const toParse = open(projectHome + "/Mason.toml", iomode.r);
+    var tomlFile = new owned(parseToml(toParse));
+    const name = tomlFile['brick']['name'].s;
+    return name;
+    }
+  catch {
+    here.chdir(MASON_HOME);
+    rmTree('/tmp');
+    writeln('Error getting the name of your package, ensure your package is a mason project');
+    exit(1);
+    }
+}
+
+/* Adds package to the Bricks of the mason-registry branch and then vrares the version.toml
+ with the source url of the package's GitHub repo.*/
+proc addPackageToBricks(projectLocal: string) : string {
+  const url = gitUrl();
+  const cwd = getEnv("PWD");
+  const toParse = open(projectLocal+ "/Mason.toml", iomode.r);
+  var tomlFile = new owned(parseToml(toParse));
+  const packageName = tomlFile['brick']['name'].s;
+  const versionNum = tomlFile['brick']['version'].s;
+  const oldDir = here.cwd();
+  here.chdir(MASON_HOME + "/tmp/mason-registry/Bricks/");
+  mkdir(packageName);
+  here.chdir(MASON_HOME + "/tmp/mason-registry/Bricks/" + packageName + "/");
+  const baseToml = tomlFile;
+  var newToml = open(versionNum + ".toml", iomode.cw);
+  var tomlWriter = newToml.writer();
+  baseToml["brick"]["source"] = url(1..url.length-1);
```

Comment:
I personally prefer array/string-indexing with brackets, but it's ultimately up to you:

```suggestion
  baseToml["brick"]["source"] = url[1..url.length-1];
```